### PR TITLE
Add Helper Classes to Parse Statement Types

### PR DESCRIPTION
**Pull request by dzou**
This adds the initial helper classes to parse statement types.

Contributes to DML and DDL efforts in which statement type must be known.

### CODE REVIEW

File: src/main/java/com/google/cloud/spanner/r2dbc/statement/StatementParser.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+      return StatementType.DDL;
+    } else if (isDml(processedSql)) {
+      return StatementType.DML;
+    } else {
+      return StatementType.UNKNOWN;
+    }
+  }
+
+  private static boolean isSelectQuery(String sql) {
+    return statementStartsWith(sql, SELECT_STATEMENTS);
```
Comment (dmitry-s) [inline src/main/java/com/google/cloud/spanner/r2dbc/statement/StatementParser.java]: maybe just check in place without these one-line methods?

File: src/main/java/com/google/cloud/spanner/r2dbc/SpannerStatement.java [hunk 2]
```diff
@@ -140,53 +142,32 @@ public Statement bindNull(int i, Class<?> type) {
...
-        })
-        .next();
-  }
-
-  /**
-   * Returns whether we think the query is a DML query.
-   */
-  private static boolean isDmlQuery(PartialResultSet firstPartialResultSet) {
-    return firstPartialResultSet.getMetadata().getRowType().getFieldsList().isEmpty();
+    StatementType statementType = StatementParser.getStatementType(this.sql);
```
Comment (elefeint) [inline src/main/java/com/google/cloud/spanner/r2dbc/SpannerStatement.java]: Can doing it multiple times (in `execute` and `runSingleStatement()`, which gets run multiple times for each set of parameters) be avoided? Perhaps pass to `runSingleStatement()` in a parameter.

File: src/main/java/com/google/cloud/spanner/r2dbc/statement/StatementParser.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+  private static final List<String> DDL_STATEMENTS = Arrays.asList("create", "drop", "alter");
+  private static final List<String> DML_STATEMENTS = Arrays.asList("insert", "update", "delete");
+
+  private StatementParser() {}
+
+  /**
+   * Returns the statement type of a given SQL string.
+   */
+  public static StatementType getStatementType(String sql) {
+    String processedSql = sql.trim().toLowerCase();
```
Comment (elefeint) [inline src/main/java/com/google/cloud/spanner/r2dbc/statement/StatementParser.java]: I'm still iffy on lowercasing the whole string. Queries can get large.

File: src/main/java/com/google/cloud/spanner/r2dbc/statement/StatementParser.java (hunk 1 above)
Comment (dzou) [reply src/main/java/com/google/cloud/spanner/r2dbc/statement/StatementParser.java]: Done.

File: src/main/java/com/google/cloud/spanner/r2dbc/SpannerStatement.java (hunk 2 above)
Comment (dzou) [reply src/main/java/com/google/cloud/spanner/r2dbc/SpannerStatement.java]: Done.

File: src/main/java/com/google/cloud/spanner/r2dbc/statement/StatementParser.java (hunk 3 above)
Comment (dzou) [reply src/main/java/com/google/cloud/spanner/r2dbc/statement/StatementParser.java]: Yeah I agree. Sounds good, I can hold off on optimizing it for now; we can figure out a better method in the future.

**Comment by dzou** [comment]
> We should probably take care of [ table_hint_expr ][ join_hint_expr ] as they can be present in the beginning of the query.
> See https://cloud.google.com/spanner/docs/query-syntax

Done.

**Pull request merged**